### PR TITLE
Add github actions CI flow

### LINKED ISSUE #1414: Setup github actions for CI (was: Travis-CI alternatives)
Travis-CI has been amazing for Dex, but was recently acquired and undergone layoffs.

https://twitter.com/carmatrocity/status/1098538649908666368

Consider switching to another CI system.

### PR DESCRIPTION
**Pull request by sagikazarmark**
This is a first version.

I plan to do further improvements around test running in subsequent PRs:

- Improve linting strategy
- Improve Kubernetes integration tests (following practices from controller-runtime)

Fixes #1414
Fixes #1557

**Update:** I've rewritten the LDAP tests to use docker and https://github.com/osixia/docker-openldap instead, because I couldn't get it working otherwise (hence the giant changeset).

### CODE REVIEW

File: connector/ldap/testdata/core.schema [hunk 1]
```diff
@@ -1,610 +0,0 @@
-# OpenLDAP Core schema
```
Comment (bonifaido) [inline connector/ldap/testdata/core.schema]: Weren't these files used at all?

Comment (sagikazarmark) [reply connector/ldap/testdata/core.schema]: The earlier tests started slapd with a clean config. These schemas are already loaded into the docker version.

Comment (bonifaido) [reply connector/ldap/testdata/core.schema]: 👍

**Comment by srenatus** [comment]
🎉 thanks, this is awesome.

**Pull request merged**